<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>
  <arg name="image" default="/usb_cam/image_raw" />

  <!-- Config and weights folder. -->
  <arg name="yolo_weights_path"          default="$(find block_device)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find block_device)/yolo_network_config/cfg"/>

  <!-- ROS and network parameter files -->
  <arg name="ros_param_file"             default="$(find block_device)/config/ros.yaml"/>
  <arg name="network_param_file"         default="$(find block_device)/config/yolov2-tiny.yaml"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="block_device_LD3" file="$(arg ros_param_file)"/>
  <rosparam command="load" ns="block_device_LD3" file="$(arg network_param_file)"/>

  <!-- Start darknet and ros wrapper -->
  <node pkg="block_device" type="block_device_LD3" name="block_device_LD3" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <remap from="/usb_cam/image_raw"  to="$(arg image)" />
  </node>

 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->
</launch>
